odoo/addons/base/res/res_partner.py

```diff
@@ -624,6 +624,11 @@ def name_create(self, name):
             If only an email address is received and that the regex cannot find
             a name, the name will have the email value.
             If 'force_email' key in context: must find the email address. """
+        context = dict(self._context)
```

Comment:
this line can be moved under the if (no need to build a dict we use only if the condition is met)